File: 2017_May23/performance/primes/GenPrime.java
import java.math.*;

class GenPrime {
    public static final long NUM_PRIMES = 100000;

    public static boolean isPrime(long v) {
        long s = ((long)Math.sqrt(v)) + 1;

        for (long i=2; i<s; i++) {
            if ((v%i) == 0)
                return false;
        }
        return true;
    }

    public static void genPrimes(long n) {
        int i = 2;
        
        while (n > 0) {
            if (isPrime(i)) {
                System.out.println(i);
                --n;
            }
            ++i;
        }
    }

    public static void main(String[] args) {
        System.out.printf("Generating %d prime numbers.\n", NUM_PRIMES);

        long start = System.currentTimeMillis();
        genPrimes(NUM_PRIMES);    
        long duration = System.currentTimeMillis() - start;

        System.out.printf("Duration: %d.%d seconds.\n",
                            (duration / 1000), (duration % 1000));
    }

}

File: 2017_May23/performance/primes/gen_prime.c
#include <stdio.h>
#include <sys/time.h>
#include <math.h>

#define NUM_PRIMES 1000

struct timeval start, end;

int is_prime(long long n) {
    long long i;
    
    long long s = (long long)sqrtl(n) + 1;

    for (i=2; i<s; i++) {
        if ((n%i) == 0) 
            return 0;
    }
    return 1;
}

void gen_primes(int num_primes)
{
    long long i = 2;

    while (num_primes) {
        if (is_prime(i)) { 
            printf("%lld\n", i);
            --num_primes;
        }
        ++i;
    }
}

int main()
{
    printf("Generating first %d prime numbers...\n", NUM_PRIMES);
    gettimeofday(&start, NULL);
    gen_primes(100000);
    gettimeofday(&end, NULL);

    printf("Duration: %d.%d seconds.\n", 
                (int)(end.tv_sec - start.tv_sec),
                (int)(end.tv_usec - start.tv_usec));

    return 0;
}

